vulnerabilities/pipelines/v2_improvers/unfurl_version_range.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import logging
+from traceback import format_exc as traceback_format_exc
+
+from aboutcode.pipeline import LoopProgress
+from fetchcode.package_versions import SUPPORTED_ECOSYSTEMS as FETCHCODE_SUPPORTED_ECOSYSTEMS
+from packageurl import PackageURL
+from univers.version_range import RANGE_CLASS_BY_SCHEMES
+from univers.version_range import VersionRange
+
+from vulnerabilities.models import ImpactedPackage
+from vulnerabilities.models import PackageV2
+from vulnerabilities.pipelines import VulnerableCodePipeline
+from vulnerabilities.pipes.fetchcode_utils import get_versions
+from vulnerabilities.utils import update_purl_version
+
+
+class UnfurlVersionRangePipeline(VulnerableCodePipeline):
+
+    pipeline_id = "unfurl_version_range_v2"
+
+    @classmethod
+    def steps(cls):
+        return (cls.unfurl_version_range,)
+
+    def unfurl_version_range(self):
+        impacted_packages = ImpactedPackage.objects.all().order_by("-created_at")
+        impacted_packages_count = impacted_packages.count()
+
+        processed_impacted_packages_count = 0
+        processed_affected_packages_count = 0
+        cached_versions = {}
+        self.log(f"Unfurl affected vers range for {impacted_packages_count:,d} ImpactedPackage.")
+        progress = LoopProgress(total_iterations=impacted_packages_count, logger=self.log)
+        for impact in progress.iter(impacted_packages):
+            purl = PackageURL.from_string(impact.base_purl)
+            if not impact.affecting_vers or not any(
+                c in impact.affecting_vers for c in ("<", ">", "!")
```

Comment:
Won't this ignore:
vers like this?
=1.0.0 or ^1.2.0

Reply:
No, this is a vers string, and VERS spec only allows `>`, `<`, and `!` for comparison.